org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/MavenProjectImporter.java

```diff
@@ -189,9 +188,9 @@ public IStatus runInWorkspace(IProgressMonitor monitor) throws CoreException {
 					if (monitor.isCanceled()) {
 						return Status.CANCEL_STATUS;
 					}
-					JavaLanguageServerPlugin.logInfo("Updating project configuration for Maven project " + project.getName());
-					MavenUpdateRequest request = new MavenUpdateRequest(project, MavenPlugin.getMavenConfiguration().isOffline(), true);
-					configurationManager.updateProjectConfiguration(request, monitor);
+					// maven update should not be forced here, because it's a new project, just imported into workspace and
+					// will be run automatically, because pom.xml-s have just appeared
+					new MavenBuildSupport(configurationManager).update(project, monitor);
```

Comment:
move MavenBuildSupport instantiation out of the loop

Reply:
Why not do projectsManage.getBuildSupport(project).update(project, monitor)? The assumption we have is that MavenBuildSupport is stateless, but I don't really like that.